pandas/plotting/_misc.py

```diff
@@ -12,18 +12,22 @@ def table(ax, data, rowLabels=None, colLabels=None, **kwargs):
     ax : Matplotlib axes object
     data : DataFrame or Series
         Data for table contents.
-    **kwargs
-        Keyword arguments to be passed to matplotlib.table.table.
+    rowLabels : list of str, optional
```

Comment:
can you type the inputs (that were added and others if you can)

Reply:
@jreback Sorry, I don't understand what I need to do
The simplest test is like this:
```
import pandas as pd
from matplotlib import pyplot as plt

df = pd.DataFrame({'old_title':[42,73]})
fig,ax = plt.subplots(1,1)
pd.plotting.table(ax,df, colLabels=['new_title'], loc='center')
ax.axis('off')
plt.show()
```
One should see a table with column named "new_title" instead of "old_title" after the PR been applied

Reply:
There are test modules in ``pandas/tests/plotting``.
In particular, I guess that your tests should go into ``pandas/tests/plotting/frame/test_frame.py``.
You can check that there is one test function on ``table`` functionality, called ``test_table``.
We use a helper function ``_check_plot_works`` for that.

At a minimum you may create a test function similar to ``test_table``, but with colLabels and rowLabels set explicitly.
Of course it would be better to check that ``matplotlib.pyplot.table`` was called with these kwargs.
For this you can mock ``matplotlib.pyplot.table`` and assert that it was called with the correct args and kwargs.

Reply:
@ivanovmg Thanks, I've dived into the pandas testing code and the testing guideline. But there are some new questions, now, sorry)

> We use a helper function _check_plot_works for that.
>
> At a minimum you may create a test function similar to test_table, but with colLabels and rowLabels set explicitly.

Am I right, that this part of your proposal just implies to use _check_plot_works in a new test function in a way like this:
`_check_plot_works(pd.plotting.table, data=df, colLabels=['new_title'], row_labels=['row1','row2'])`
for a df i've provided above, for instance?

> Of course it would be better to check that matplotlib.pyplot.table was called with these kwargs.
> For this you can mock matplotlib.pyplot.table and assert that it was called with the correct args and kwargs.

But how to do this? How to "mock `matplotlib.pyplot.table`" and how to assert what args and kwargs it was called with?

Reply:
> Am I right, that this part of your proposal just implies to use _check_plot_works in a new test function in a way like this:
> `_check_plot_works(pd.plotting.table, data=df, colLabels=['new_title'], row_labels=['row1','row2'])`
> for a df i've provided above, for instance?

Yes, looks like this.
You may also want to add a test when you call ``df.plot`` with kwarg ``table=True``, which I believe should result in the same outcome as ``pd.plotting.table``, shouldn't it?

> > Of course it would be better to check that matplotlib.pyplot.table was called with these kwargs.
> > For this you can mock matplotlib.pyplot.table and assert that it was called with the correct args and kwargs.
> 
> But how to do this? What do you mean saying to "mock `matplotlib.pyplot.table`" and how to assert what args and kwargs it was called with?

Probably something like this (I did not check it would work though, so please try and update as necessary).

```
from unittest.mock import patch
with patch("matplotlib.pyplot.table", return_value=None) as mock_table:
    df.plot(table=True, colLabels=["new_title"])
    kwargs = mock_table.call_args.kwargs
    assert kwargs["colLabels"] == ["new_title"]

```
Reference: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_args

Reply:
Cool! I'll try this and will provide the code, thank you. It may take time a little